src/lib/es2015.iterable.d.ts

```diff
@@ -497,13 +553,20 @@ interface Float64Array {
 }
 
 interface Float64ArrayConstructor {
-    new (elements: Iterable<number>): Float64Array;
+    new (elements: Iterable<number>): Float64Array<ArrayBuffer>;
 
     /**
      * Creates an array from an array-like or iterable object.
      * @param arrayLike An array-like or iterable object to convert to an array.
      * @param mapfn A mapping function to call on every element of the array.
      * @param thisArg Value of 'this' used to invoke the mapfn.
      */
-    from(arrayLike: Iterable<number>, mapfn?: (v: number, k: number) => number, thisArg?: any): Float64Array;
+    from(arrayLike: Iterable<number>): Float64Array<ArrayBuffer>;
+    /**
+     * Creates an array from an array-like or iterable object.
+     * @param arrayLike An array-like or iterable object to convert to an array.
+     * @param mapfn A mapping function to call on every element of the array.
+     * @param thisArg Value of 'this' used to invoke the mapfn.
+     */
+    from<T>(arrayLike: Iterable<T>, mapfn?: (v: T, k: number) => number, thisArg?: any): Float64Array<ArrayBuffer>;

```

Comment:
How breaky is this change in particular? I'm guessing if people are calling e.g.  `int32array.from(iterable, (...) => { ... })`, we'd pretty much always be able to infer something good for `T`, but I'm no inference expert.

Reply:
This shouldn't break any existing code. Any place that previously would have passed an iterable or arraylike for `number` will still get `number` values. This is actually more flexible than what we had, and was already being used on BigInt64Array.